_posts/2016-06-28-tips-for-upgrading-rails-3-2-to-4.markdown

```diff
@@ -0,0 +1,90 @@
+---
+layout: post
+title: "Tips for upgrading from Rails 3.2 to 4.0"
+date: 2016-06-28 11:08:00
+categories: ["rails"]
+author: "mauro-oto"
+---
+
+There are already quite a few guides in the wild to help with the upgrade of
+Rails 3.2 to Rails 4.0.
+The [official Rails guide](http://edgeguides.rubyonrails.org/upgrading_ruby_on_rails.html#upgrading-from-rails-3-2-to-rails-4-0)
+for upgrading from Rails 3.2 to 4.0 is very thorough.
+With the upcoming release of Rails 5.0, apps currently in production running
+Rails 3.2 should probably be updated to any stable Rails 4 release as soon as
+possible.
+
+Even an e-book on [upgrading Rails 3 to 4](https://github.com/alindeman/upgradingtorails4)
+exists, which serves as a useful guide to make this upgrade easier and helps
+understand the advantages & disadvantages of this new (soon to be old) version.
+
+However, if you're using any non-standard gems, you're mostly on your own. Some
+gems stopped being maintained before Rails 4 was released, as was the case
+with [CanCan](https://github.com/ryanb/cancan), a well known authorization
+library. After many open pull requests were left [unmerged](https://github.com/ryanb/cancan/pulls),
+[CanCanCan](https://github.com/CanCanCommunity/cancancan) was released.
+It is a community driven effort to have a semi-official fork of CanCan.
+It serves as a drop-in replacement for those wanting to keep CanCan and upgrade
```

Comment:
@mauro-oto "those wanting" sounds weird. I'd change it to: It serves as a drop-in replacement for people who want to use CanCan after upgrading to Rails 4.